builtin/iter_test.mbt

```diff
@@ -242,6 +242,52 @@ test "collect" {
   @assertion.assert_eq(vec, ['1', '2', '3', '4', '5'])?
 }
 
+test "range" {
+  // edge cases
+  inspect(
+    (0).to(10).take(100).collect(),
+    content="[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]",
+  )?
+  inspect((100).to(10).take(100).collect(), content="[]")?
+  // to
+  inspect(
+    (0).to(10).take(5).concat((0).to(2)).collect(),
+    content="[0, 1, 2, 3, 4, 0, 1, 2]",
+  )?
+  inspect(
+    0x80000000L.to(0x80000000L + 10L).take(5).concat(
+      (0x80000000L + 10L).to(0x80000000L + 20L).take(3),
+    ).collect(),
+    content="[2147483648, 2147483649, 2147483650, 2147483651, 2147483652, 2147483658, 2147483659, 2147483660]",
+  )?
+  inspect(
+    0.0.to(10.0).take(5).concat(0.0.to(2.0)).collect(),
+    content="[0.0, 1.0, 2.0, 3.0, 4.0, 0.0, 1.0, 2.0]",
+  )?
+  inspect(
+    (-50).to(50, step=10).take(100).collect(),
+    content="[-50, -40, -30, -20, -10, 0, 10, 20, 30, 40, 50]",
+  )?
+  inspect(
+    0.0.to(1.0, step=0.1).take(5).collect(),
+    content="[0.0, 0.1, 0.2, 0.30000000000000004, 0.4]",
+  )?
+  inspect(
+    0x80000000L.to(0x80000000L + 10L, step=2L).take(5).collect(),
+    content="[2147483648, 2147483650, 2147483652, 2147483654, 2147483656]",
```

Comment:
let's make it `[a,b)`

Reply:
My concern is, 0 to 10 sounds like `0-10`, and this is also what ruby does.
But exclusive end should be better for programming though.
<img width="943" alt="Screenshot 2024-06-17 at 16 51 47" src="https://github.com/moonbitlang/core/assets/29861494/0646d1ec-a807-400b-9bb8-45ed07dcf68e">